pkg/volume/csi/csi_plugin.go

```diff
@@ -440,9 +440,9 @@ func (p *csiPlugin) NewMounter(
 
 	if err := saveVolumeData(dataDir, volDataFileName, volData); err != nil {
 		klog.Error(log("failed to save volume info data: %v", err))
-		if err := os.RemoveAll(dataDir); err != nil {
-			klog.Error(log("failed to remove dir after error [%s]: %v", dataDir, err))
-			return nil, err
+		if removeErr := os.RemoveAll(dataDir); removeErr != nil {
+			klog.Error(log("failed to remove dir after error [%s]: %v", dataDir, removeErr))
+			return nil, removeErr
```

Comment:
Good catch, however, would it make sense to return err here (and below in NewBlockVolumeMapper)? Because that's why setup failed.

Reply:
Thanks, maybe return `err` is better. 

I think we should return why it failed initially. I have resent the  PR, PTAL!

Reply:
This pr would work when `saveVolumeData` failed but `RemoveAll` succeeded. Then we would return  `nil, nil`

This may cause panic since some of the callers do not care if the first returned value is `nil` or not, for example:

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go:L1027

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go:L590